executor.go

```diff
@@ -180,14 +200,32 @@ func (e *ExecutorBuilder) NewCmdExecutor(ctx context.Context) cmd.Executor {
 			logger.Warning("[SERVICE: Bloomfilter]", err.Error())
 		}
 
-		pf := e.ProxyFactory.NewProxyFactory(
-			logger,
-			e.BackendFactory.NewBackendFactory(ctx, logger, metricCollector),
-			metricCollector,
-		)
+		var bpf proxy.BackendFactory
+		if bfwc, ok := e.BackendFactory.(BackendFactoryWithConfig); ok {
+			bpf = bfwc.NewBackendFactoryWithConfig(ctx, logger, metricCollector, &cfg)
+		} else {
+			bpf = e.BackendFactory.NewBackendFactory(ctx, logger, metricCollector)
+		}
+
+		var pf proxy.Factory
+		if pfwc, ok := e.ProxyFactory.(ProxyFactoryWithConfig); ok {
+			pf = pfwc.NewProxyFactoryWithConfig(logger, bpf, metricCollector, &cfg)
+		} else {
+			pf = e.ProxyFactory.NewProxyFactory(logger, bpf, metricCollector)
+		}
 
 		agentPing := make(chan string, len(cfg.AsyncAgents))
 
+		handlerF := e.HandlerFactory.NewHandlerFactory(logger, metricCollector, tokenRejecterFactory)
+		otelCfg, err := otelconfig.FromLura(cfg)
+		if err == nil {
+			handlerF = otelgin.New(handlerF, otelstate.GlobalState, otelCfg.SkipPaths)
+		}
+
+		runServerChain := router.RunServerFunc(e.RunServerFactory.NewRunServer(logger, serverhttp.RunServerWithLoggerFactory(logger)))
+		if otelCfg != nil {
```

Comment:
@dhontecillas ideally shouldn't the first handler have instrumentation? In this case cors middleware execution results won't be captured in the telemetry data right?

Reply:
yep! I am going to fix it .. thanks